company-lsp.el

```diff
@@ -139,6 +140,24 @@ Returns CANDIDATE with the resolved CompletionItem."
                              candidate)))))
   candidate)
 
+(defun company-lsp--expand-arguments (item)
+  "Expand the function's arguments.
+ITEM is a CompletionItem."
+  (let* ((kind (gethash "kind" item))
+	 (is-function (when (numberp kind) (= kind 3)))
+	 (language-id (lsp--workspace-language-id lsp--cur-workspace))
+	 (detail (gethash "detail" item)))
+    (when (and (fboundp 'yas-expand-snippet)
+	       (string= language-id "rust")
```

Comment:
How about we make this more general and customizable? We can have an alisp `company-lsp-snippet-functions` with the key of language ID and the value of a function that takes a completion item and returns nil or a snippet string. Then have a function `company-lsp-rust-completion-snippet` for rust language server.